torch/csrc/jit/codegen/cuda/predicate_compute.cpp

```diff
@@ -103,10 +105,24 @@ kir::Bool* PredicateCompute::getInlinePredicate(
   auto pred_inds =
       Index::getConsumerRootPredIndices(out_tv, loops, pred_contiguity);
   auto root_indices = pred_inds.first;
-  bool use_rfactor = pred_inds.second;
+  bool use_maybe_rfactor = pred_inds.second;
+
+  if (out_tv->getMemoryType() != MemoryType::Global && out_tv->hasReduction() &&
```

Comment:
I think we should not skip predicates for shared memory buffers as well. No predicate would mean shared memory buffers would be zero-initialized by all threads redundantly, right? That would cause race conditions, then.

Reply:
I didn't think about it, yeah that seems fair. Could you please commit that change and then go ahead and merge this if you approve otherwise?

Reply:
Thanks for the commit.